torch/nn/parallel/distributed.py

```diff
@@ -254,12 +253,16 @@ def forward(ctx, reducer, ddp_state, *inputs):
     def backward(ctx, *grad_outputs):
         # Enqueue delay allreduce for static graph training on the first
         # iteration.
-        if ctx.ddp_state["static_graph"] and ctx.ddp_state["num_forward_calls"] == 1:
+        ddp_weakref = ctx.ddp_weakref()
```

Comment:
nit: In my understanding, after you call the weakref (right-hand-side), you retrieve the original object, which should now be a strong reference? Maybe the left-hand-side should be named `ddp`?